core/dbt/cli/flags.py

```diff
@@ -63,6 +97,12 @@ def assign_params(ctx, params_assigned_from_default):
             for param_assigned_from_default in params_assigned_from_default:
                 user_config_param_value = getattr(user_config, param_assigned_from_default, None)
                 if user_config_param_value is not None:
+                    # TODO Conversion should happen here to make sure types are consistent with the flag
+                    # extract into a function
+                    if param_assigned_from_default in ["warn_error_options"]:
+                        user_config_param_value = WarnErrorOptionsType().convert(
+                            user_config_param_value, None, None
```

Comment:
`warn_error_options` is a [dictionary](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/contracts/project.py#L252) when loaded via `UserConfig`, but `WarnErrorOptionsType().convert` expects a string, which leads to runtime errors when `warn_error_options` is set in the user config. 

For example: 
```yaml
#profiles.yml
config:
  warn_error_options:
    exclude:
    - PackageInstallPathDeprecation
    include: all
```

```sh
❯ python3 core/dbt/cli/main.py run --project-dir ~/src/jaffle_shop
Usage: main.py run [OPTIONS]
Try 'main.py run -h' for help.

Error: Invalid value: Cannot load YAML from type <class 'dict'>
```

I think our options would be to either: 
1. Pass `'include'` and `'exclude'` keys from from `user_config_param_value` dict to tthe `WarnErrorOptions` constructor directly instead of going through `WarnErrorOptionsType`, or 
2. convert `user_config_param_value` to a string representation before passing it to `WarnErrorOptionsType().convert`

My preference is the first option so we don't go through an unnecessary series of str -> dict -> str conversions assuming there are no circular dependency issues in using `WarnErrorOptions` directly.

Reply:
Adjusted things via the first way and moved conversion for original args and user config to a single place.